From 3f682a36dec5f2f6277190564c0fe61711ed9164 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Thu, 27 Apr 2023 13:18:20 +0800
Subject: [PATCH 64/68] Revert "drm/virtio: Fix NULL vs IS_ERR checking in
 virtio_gpu_object_shmem_init"

This reverts commit 0a4181b23acf53e9c95b351df6a7891116b98f9b.
---
 drivers/gpu/drm/virtio/virtgpu_object.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/virtio/virtgpu_object.c b/drivers/gpu/drm/virtio/virtgpu_object.c
index d4fab3361d2c3..0c98978e2e55c 100644
--- a/drivers/gpu/drm/virtio/virtgpu_object.c
+++ b/drivers/gpu/drm/virtio/virtgpu_object.c
@@ -157,9 +157,9 @@ static int virtio_gpu_object_shmem_init(struct virtio_gpu_device *vgdev,
 	 * since virtio_gpu doesn't support dma-buf import from other devices.
 	 */
 	shmem->pages = drm_gem_shmem_get_sg_table(&bo->base.base);
-	if (IS_ERR(shmem->pages)) {
+	if (!shmem->pages) {
 		drm_gem_shmem_unpin(&bo->base.base);
-		return PTR_ERR(shmem->pages);
+		return -EINVAL;
 	}
 
 	if (use_dma_api) {
-- 
2.25.1

